bigchaindb/core.py

```diff
@@ -553,41 +556,47 @@ def vote(self, block, previous_block_id, decision, invalid_reason=None):
 
         return vote_signed
 
-    def write_vote(self, block, vote, block_number):
+    def write_vote(self, block, vote):
         """Write the vote to the database."""
 
         # First, make sure this block doesn't contain a vote from this node
         if self.has_previous_vote(block):
             return None
 
-        # We need to *not* override the existing block_number, if any
-        # FIXME: MIGHT HAVE RACE CONDITIONS WITH THE OTHER NODES IN THE FEDERATION
-        if 'block_number' not in vote:
-            vote['block_number'] = block_number  # maybe this should be in the signed part...or better yet, removed..
-
         r.table('votes') \
             .insert(vote) \
             .run(self.conn)
 
     def get_last_voted_block(self):
         """Returns the last block that this node voted on."""
 
+
         last_voted = r.table('votes') \
```

Comment:
The problem is we want to return all newest timestamps so that we can then ordering following the chain. In your example with min we would want it to return `['a', 'a']`